atc/engine/check_delegate.go

```diff
@@ -111,25 +118,56 @@ func (d *checkDelegate) WaitToRun(ctx context.Context, scope db.ResourceConfigSc
 		}
 	}
 
-	end, err := scope.LastCheckEndTime()
-	if err != nil {
-		if releaseErr := lock.Release(); releaseErr != nil {
-			logger.Error("failed-to-release-lock", releaseErr)
+	shouldRun := false
+	if !d.plan.IsPeriodic() {
+		lastCheckEndTime, lastCheckSucceeded, err := scope.LastCheckEndTime()
```

Comment:
querying the last check start time/end time/succeeded seems to be making this logic pretty noisy and hard to follow - wdyt of putting it one call above all the branching, since we can fetch all this in one query pretty cheaply, and we need at least some of it in all but one of these branches?

```go
lastCheck, err := scope.LastCheck()
// ...

lastCheck.StartTime
lastCheck.EndTime
lastCheck.Succeeded
```

Reply:
Good idea.